### PR TITLE
Fix PyNEST test failures

### LINKED ISSUE #6: PyNEST tests using AssertGreater fail
All PyNEST tests using the compatibility versions of `AssertGreater` fail with older versions of Python and SciPy (here Python 2.6.6, SciPy 0.7.2). The error messages look like this:

```
======================================================================
ERROR: testRPortDistribution (nest.tests.test_connect_all_patterns.TestAllToAll)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/eppler/10kcollaps.build/install/lib64/python2.6/site-packages/nest/tests/test_connect_all_to_all.py", line 88, in testRPortDistribution
    self.assertGreater(p, self.pval, 'Chi2 test failed.')
TypeError: <lambda>() takes exactly 3 arguments (4 given)

======================================================================
ERROR: testRPortDistribution (nest.tests.test_connect_all_to_all.TestAllToAll)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/eppler/10kcollaps.build/install/lib64/python2.6/site-packages/nest/tests/test_connect_all_to_all.py", line 88, in testRPortDistribution
    self.assertGreater(p, self.pval, 'Chi2 test failed.')
TypeError: <lambda>() takes exactly 3 arguments (4 given)
```

### PR DESCRIPTION
**Pull request by jougs**
This PR does three things compared to test failures with older Python and NumPy versions:
* Check if the nosetests executable supports the `--with-xunit` switch. If it does not, we use our own test harness instead.
* Remove the error message given to `assertGreater`, as this leads to problems with Python <= 2.6. This fixes #6 .
* Add a simple implementation for `numpy.fill_diagonal` if it is not there (NumPy <= 1.4.0)

### CODE REVIEW

File: pynest/nest/tests/compatibility.py [hunk 1]
```diff
@@ -65,3 +66,10 @@ def decorator(test_item):
 if not hasattr(unittest.TestCase, 'assertGreater'):
     unittest.TestCase.assertGreater = lambda self, a, b, **kwargs: \
         unittest.TestCase.assertTrue(self, a > b, **kwargs)
+
+# NumPy 1.4.0-
+if not hasattr(numpy, 'fill_diagonal'):
+    def fill_diagonal(a, b):
```
Comment (heplesser) [inline pynest/nest/tests/compatibility.py]: Do we need to handle non-square matrices explicitly?

**Comment by heplesser** [comment]
@jougs Fine, but I wonder if we need to protect against special cases, see above. Otherwise, :+1:.

File: pynest/nest/tests/compatibility.py [hunk 2]
```diff
@@ -65,3 +66,10 @@ def decorator(test_item):
...
+        diagonal = numpy.arange(a.shape[0])
+        a[diagonal, diagonal] = b
```
Comment (tammoippen) [inline pynest/nest/tests/compatibility.py]: You could write those two lines as follows, to handle matrices of all shapes:

    diagonal = [ numpy.arange( min( numpy.shape( a ) ) ) ] * len( numpy.shape( a ) )
    a[ diagonal ] = b

**Comment by tammoippen** [comment]
The PR looks fine to me. After @jougs commented on the special cases mentioned, I am good to give my :+1: .

**Comment by jougs** [comment]
@heplesser, @tammoippen: I've added a generalized version for arbitrarily shaped matrices.

File: pynest/nest/tests/compatibility.py (hunk 2 above)
Comment (jougs) [inline pynest/nest/tests/compatibility.py]: See new commit cde542f below.

**Comment by heplesser** [comment]
:+1:

**Comment by tammoippen** [comment]
Nice. Merging.

**Pull request merged**